### PR TITLE
ISSUE-309: Grid state

### PR DESCRIPTION
**Pull request by joneit**
## WIP

See issue https://github.com/openfin/fin-hypergrid/issues/309 for details.

Task list:
- [x] `rows`
- [x] `columns`
- [x] `cells`
- [ ] `columnNames`
- [ ] `plugins`
- [ ] `subgrids`
- [ ] `calculators`

### CODE REVIEW

**Comment by Dwaynekj** [comment]
With regards to https://github.com/openfin/fin-hypergrid/issues/309

**Comment by Dwaynekj** [comment]
@joneit 
Can we split demo.js up into several files one for:
* theme.js / dynamic properties
* state.js

We should also make it clear that all the grid's dynamic properties should be set from  `lib.dynamic`

**Comment by Dwaynekj** [comment]
@joneit When you get back to this ticket, please review this issue https://github.com/openfin/fin-hypergrid/issues/573

**Comment by joneit** [comment]
Rebased to develop.

**Comment by joneit** [comment]
@Dwaynekj Suggest you merge in only the changes I just made to the following files:

* `.gitignore`
* `gulpfile.js`

And then delete `demo/js/demo/build` folder from github.

File: demo/js/demo/setState.js [hunk 1]
```diff
@@ -186,4 +185,12 @@ module.exports = function (grid) {
     //grid.addProperties(myThemes.three);
 
     grid.takeFocus();
+    //see myThemes.js file for how to create a theme
+    //grid.addProperties(myThemes.one);
+    //grid.addProperties(myThemes.two);
+    //grid.addProperties(myThemes.three);
+
+    grid.takeFocus();
```
Comment (Dwaynekj) [inline demo/js/demo/setState.js]: you added takeFocus twice

**Pull request merged**